mito-ai/src/Extensions/AiChat/ChatHistoryManager.tsx

```diff
@@ -203,6 +203,7 @@ export class ChatHistoryManager {
 
         const agentExecutionMetadata: IAgentExecutionMetadata = {
             promptType: 'agent:execution',
+            activeCellId: getActiveCellID(this.notebookTracker) || '',
```

Comment:
Hmm I think this is going to return the wrong results when the user switches notebooks while the agent is working. So instead of using the notebookTracker, let's use the notebookPanel that we pass to the function. The notebookPanel is the agentTargetNotebook which we store whenever a new agent execution starts.